core/modules/qproc/QuerySession.cc

```diff
@@ -356,21 +354,25 @@ void QuerySession::_applyConcretePlugins() {
 }
 
 /// Some code useful for debugging.
-void QuerySession::_showFinal(std::ostream& os) {
-    // Print out the end result.
-    query::QueryTemplate par = _stmtParallel.front()->getTemplate();
-    query::QueryTemplate mer = _stmtMerge->getTemplate();
-
-    os << "\tQuerySession::_showFinal() : parallel: " << par.dbgStr() << std::endl;
-    os << "\tQuerySession::_showFinal() : merge: " << mer.dbgStr() << std::endl;
+std::string QuerySession::toString() {
+    std::ostringstream os;
+    query::QueryTemplate par = _stmtParallel.front()->getQueryTemplate();
+    query::QueryTemplate mer = _stmtMerge->getQueryTemplate();
+    os << "QuerySession description:" << std::endl;
```

Comment:
Do not use `std::endl` unless you really mean to flush the stream, `\n` is sufficient here.